### ISSUE TITLE
True color (24-bit) not working

### ISSUE THREAD

**Issue by thallada** [rpwoodbu/mosh-chrome#188]
Thanks for this awesome app. I'm having some trouble getting true color support working.

I signed into the same server using mosh-chrome and the Secure Shell app. I used the same [true color support test script mentioned here](https://unix.stackexchange.com/a/404415) in both.

Secure Shell:
![image](https://user-images.githubusercontent.com/1505923/41502786-3334cd98-7190-11e8-87a2-c7bd2b11ca97.png)

mosh-chrome:
![image](https://user-images.githubusercontent.com/1505923/41502791-4c4f5fdc-7190-11e8-80ce-b0276f781f74.png)

In both terminals, `echo $TERM` returns `xterm-256color`.

I noticed that the version of hterm in this repo (1.6.1) is a bit behind the latest version (1.7.8). Maybe upgrading it could resolve this?

**Comment by rpwoodbu** [comment]
You may be on to something with the version of hterm. Try the "Mosh (dev)" version in the Chrome Web Store, which is newer. I'm going to be pushing hterm-1.80 tonight (in v0.5.5).

**Comment by ironiridis** [comment]
![image](https://user-images.githubusercontent.com/308159/43041738-75673dea-8d2e-11e8-9b67-ff2345cfe1d0.png)
Tested on v.0.5.5. Close?

**Comment by jsimpson-inmarsat** [comment]
Mosh (dev) v0.5.5 fails to render 24-bit color for me, appearing the same as shown in the original report above.

**Comment by jxyzn** [comment]
Mosh (dev) v0.5.5 fails to render 24-bit color for me, appearing the same as shown in the original report above.

**Comment by bukzor** [comment]
Truecolor is screwing in mosh. It seems to be the binary rather than the app though, as it's also wrong under ssh+`mosh localhost`.

Normal:
![screenshot 2018-08-22 at 1 19 14 pm](https://user-images.githubusercontent.com/640328/44490235-a7d24e80-a612-11e8-94d5-4b9ba89add6d.png)


Mosh:
![screenshot 2018-08-22 at 1 16 50 pm](https://user-images.githubusercontent.com/640328/44490222-9c7f2300-a612-11e8-8461-131c5960cc11.png)

**Comment by rpwoodbu** [comment]
They apparently added support in Mosh 1.3.2 (see mobile-shell/mosh#649), and that's the version being used in the latest dev release, but it doesn't seem to be working.

**Comment by bukzor** [comment]
1.3.2 is also what I ran in my ssh+`mosh --local 127.0.0.1` test, with same wonky result.
Here's my truecolor test https://gist.github.com/bukzor/f880fc1e93fc081f079fd6e6aed8b2ea#file-colortest16777216

**Comment by jxyzn** [comment]
Yeah, there's something else afoot. 24-bit color works in Secure Shell App 0.8.44.1, both via plain ssh and after 'mosh localhost' (with mosh 1.3.2), but does not work connected via Mosh (dev) v0.5.5 -- for me; not sure how @ironiridis got the reduced fidelity results shown in that screenshot.

**Comment by rpwoodbu** [comment]
Edited my previous statement about mosh 1.3.2. It does _not_ include the true color commit. So this is _not_ expected to work yet.

**Comment by jxyzn** [comment]
Ah yes, it takes a custom build to enable it (which I have). Wish they'd push it out officially soon.

**Comment by niutech** [comment]
I wanted to run [browsh](https://www.brow.sh) using Mosh (dev) Chrome app version 1.3.2, but it still does not display glyphs properly:

![image](https://user-images.githubusercontent.com/384997/50646025-6618c380-0f75-11e9-8cc8-678ccd61cf01.png)

When will the truecolor version be released?

**Comment by thallada** [comment]
For anyone looking for a solution to this, I was able to get true color mosh working on my chromebook with [crouton](https://github.com/dnschneid/crouton). Once crouton is set up, enter the chroot, add the [mosh-dev ppa](https://launchpad.net/~keithw/+archive/ubuntu/mosh-dev), then `sudo apt-get install mosh`. After that `mosh --version` should say "mosh 1.3.2". Then you should be able to connect to another server running mosh 1.3.2 and get true color support.

I think the crosh shell uses the Secure Shell terminal app, so it's approximately the same experience but there may be some latency since it's now going through the chroot, not sure.

I think whatever mosh client is bundled with mosh-chrome is too old or doesn't have true color support enabled.

**Comment by makeworld-the-better-one** [comment]
Also experiencing this.
> Ah yes, it takes a custom build to enable it (which I have). Wish they'd push it out officially soon.

@jxyzn How can I do this?

**Comment by jxyzn** [comment]
> @jxyzn How can I do this?

The easiest method is as @thallada describes. Either add the mosh-dev ppa, or go to the repo and grab the .deb (for me, currently http://ppa.launchpad.net/keithw/mosh-dev/ubuntu/pool/main/m/mosh/mosh_1.3.2+1354-0ppa~ubuntu19.04.1_amd64.deb) and install it.

That build is patched for 24-bit color.

**Comment by pgraham** [comment]
Is it possible to build this app against the mosh master branch rather than the release? It doesn't look like mosh is going to have another release any time soon, if ever and even the ppa suggested by @thallada hasn't been updated since Feb 2019.

**Comment by vapier** [comment]
use Secure Shell -- it includes this mosh project now.
https://chrome.google.com/webstore/detail/iodihamcpbpeioajjeobimgagajmlibd
